src/components/Animation/AnimationCanvas.vue

```diff
@@ -0,0 +1,319 @@
+          evt.context.fillStyle = "rgb(0,0,0)";
+          evt.context.fillRect(
+            0,
+            0,
+            evt.context.canvas.width,
+            evt.context.canvas.height
+          );
+          evt.context.globalCompositeOperation = "overlay";
+          evt.context.fillStyle = "rgb(0,0,0)";
+          evt.context.fillRect(
+            0,
+            0,
+            evt.context.canvas.width,
+            evt.context.canvas.height
+          );
+          evt.context.globalCompositeOperation = "difference";
+          evt.context.fillStyle =
+            "rgba(" +
+            [
+              255 - this.getRGB[0],
+              255 - this.getRGB[1],
+              255 - this.getRGB[2],
+              1.0,
+            ].toString() +
+            ")";
+          evt.context.fillRect(
+            0,
+            0,
+            evt.context.canvas.width,
+            evt.context.canvas.height
+          );
+          evt.context.globalCompositeOperation = "source-over";
+        };
+      }
+      this.$animationCanvas.mapObj
+        .getLayers()
+        .getArray()[0]
+        .on("postrender", this.darkOSMCallback);
+      this.$animationCanvas.mapObj.updateSize();
+    },
+    copyLayers() {
+      this.$mapLayers.arr.forEach((layer) => {
+        if (layer.get("layerVisibilityOn")) {
+          const originalSource = layer.getSource();
+          const originalUrl = originalSource.getUrl();
+          const originalParams = originalSource.getParams();
+          let originalProperties = Object.assign({}, layer.getProperties());
+          delete originalProperties.map;
+          delete originalProperties.source;
+
+          const newSource = new ImageWMS({
+            format: "image/png",
+            url: originalUrl,
+            params: originalParams,
+            transition: 0,
+            crossOrigin: "Anonymous",
+            ratio: 1,
+          });
+          const newLayer = new OLImage({
+            source: newSource,
+          });
+          newLayer.setProperties(originalProperties);
+          newLayer.getSource().on("imageloaderror", (e) => {
+            let layer = this.$mapLayers.arr.find(
+              (l) => l.get("layerName") === newLayer.get("layerName")
+            );
+            this.$root.$emit("loadingError", layer, e);
+            this.$root.$emit("loadingError", newLayer, e);
+          });
+          this.copiedLayers.push(newLayer);
+          this.$animationCanvas.mapObj.addLayer(newLayer);
+        }
+      });
+    },
+    handleCancelExpired() {
+      this.cancelExpired = true;
+    },
+    async mapControls() {
+      this.cancelExpired = false;
+      const driverDate =
+        this.getMapTimeSettings.Extent[this.getMapTimeSettings.DateIndex];
+      let visibleTLayers = this.copiedLayers.filter((l) => {
+        return l.get("layerVisibilityOn") && l.get("layerIsTemporal");
+      });
+      let noChange = true;
+      for (let i = 0; i < visibleTLayers.length; i++) {
```

Comment:
Likewise, change to use a `const` variable to avoid having to recalculate `.length` each loop.